### PR TITLE
networkmanager: Move network/interfaces out of the shell package

### PR DESCRIPTION
**Pull request by mvollmer**

### CODE REVIEW

File: pkg/networkmanager/interfaces.js [hunk 1]
```diff
@@ -3107,6 +3097,71 @@ function PageNetworkVlanSettings() {
     this._init();
 }
 
-shell.dialogs.push(new PageNetworkVlanSettings());
+function dialog_setup(d) {
```
Comment (stefwalter) [inline pkg/networkmanager/interfaces.js]: This section needs a comment explaining that this is porting old code, in order to explain the unexpected code that follows.

Comment (mvollmer) [inline pkg/networkmanager/interfaces.js]: Yes, good point.  Done.

File: pkg/networkmanager/Makefile.am [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+	pkg/networkmanager/interfaces.min.html.gz \
+	pkg/networkmanager/networking.min.css.gz \
+	$(NULL)
+networkmanager_DATA = \
+	pkg/networkmanager/manifest.json \
+	$(networkmanager_GZ) \
+	$(NULL)
+
+networkmanagerdebugdir = $(debugdir)$(networkmanagerdir)
+networkmanagerdebug_DATA = \
```
Comment (dperpeet) [inline pkg/networkmanager/Makefile.am]: shouldn't we have the non-minified css file in the debugdata as well?

**Comment by dperpeet** [comment]
let's find the source for that travis error

File: pkg/networkmanager/Makefile.am (hunk 2 above)
Comment (mvollmer) [inline pkg/networkmanager/Makefile.am]: Yes, you are right.  Good catch!

**Comment by mvollmer** [comment]
> let's find the source for that travis error

I can't....

**Comment by mvollmer** [comment]
About the Travis failures.

- I can't reproduce them locally.
- They go away when removing pkg/networkmanager/Makefile.am from ./Makefile.am
- They also go away when removing pkg/kubernetes/Makefile.am and leaving pkg/networkmanager/Makefile.am in place
- Updating make to the trusty version doesn't help.

I'll setup a local Ubuntu Precise VM and see what happens.

**Comment by mvollmer** [comment]
We might hit this: http://savannah.gnu.org/bugs/?33134

It seems to be fixed in GNU make 4.0, released Oct 2013.

**Comment by mvollmer** [comment]
> I'll setup a local Ubuntu Precise VM and see what happens.

Nope, can't reproduce it there either.

**Comment by stefwalter** [comment]
This is no longer blocking on Travis right?

**Comment by mvollmer** [comment]
Depends on #2611.

**Comment by mvollmer** [comment]
> Depends on #2611.

That's in master, and Travis is happy again.  Uff.

File: pkg/networkmanager/interfaces.html [hunk 3]
```diff
@@ -0,0 +1,496 @@
...
+        $(interfaces);
+    });
+</script>
+</head>
+<body hidden>
+
+  <div id="networking" class="container-fluid" hidden>
+    <div class="row">
+      <div class="col-sm-6">
+        <div>Sending<span style="float:right" id="networking-tx-text"></span></div>
```
Comment (dperpeet) [inline pkg/networkmanager/interfaces.html]: `translatable`?

File: pkg/networkmanager/interfaces.html [hunk 4]
```diff
@@ -0,0 +1,496 @@
...
+        <div style="height:120px" id="networking-tx-graph"></div>
+      </div>
+      <div class="col-sm-6">
+        <div>Receiving<span style="float:right" id="networking-rx-text"></span></div>
```
Comment (dperpeet) [inline pkg/networkmanager/interfaces.html]: `translatable`?

File: pkg/networkmanager/interfaces.html [hunk 5]
```diff
@@ -0,0 +1,496 @@
...
+        <div style="height:120px" id="networking-rx-graph"></div>
+      </div>
+    </div>
+    <br/>
+    <div class="panel panel-default"
+         id="networking-interfaces">
+      <div class="panel-heading">
+        <span translatable="yes">Interfaces</span>
+        <div style="float:right"><button id="networking-add-bond" class="btn btn-default network-privileged">Add Bond</button>
```
Comment (dperpeet) [inline pkg/networkmanager/interfaces.html]: `translatable`?

File: pkg/networkmanager/interfaces.html [hunk 6]
```diff
@@ -0,0 +1,496 @@
...
+          <button id="networking-add-bridge" class="btn btn-default network-privileged">Add Bridge</button>
+          <button id="networking-add-vlan" class="btn btn-default network-privileged">Add VLAN</button>
+        </div>
+      </div>
+      <table class="table table-hover">
+        <thead>
+          <tr>
+            <th>Name</th>
```
Comment (dperpeet) [inline pkg/networkmanager/interfaces.html]: do all these need `translatable`?

File: pkg/networkmanager/interfaces.html [hunk 7]
```diff
@@ -0,0 +1,496 @@
...
+      </div>
+    </div>
+    <br/>
+    <div class="panel panel-default">
+      <div class="panel-heading">
+        <span id="network-interface-name"></span>
+        <span style="margin-left:20px;margin-right:20px" id="network-interface-hw"></span>
+        <span id="network-interface-mac"></span>
+        <div style="float:right">
+          <button class="btn btn-danger network-privileged" style="margin-right:20px"
```
Comment (dperpeet) [inline pkg/networkmanager/interfaces.html]: `translatable`

File: pkg/networkmanager/interfaces.html [hunk 8]
```diff
@@ -0,0 +1,496 @@
...
+        <table class="cockpit-form-table" id="network-interface-settings">
+        </table>
+      </div>
+    </div>
+    <div class="panel panel-default" id="network-interface-slaves">
+      <table class="table table-hover">
+        <thead>
+          <tr>
+            <th></th>
+            <th style="width:20%">Sending</th>
```
Comment (dperpeet) [inline pkg/networkmanager/interfaces.html]: `translatable`, also next line

File: pkg/networkmanager/interfaces.html [hunk 9]
```diff
@@ -0,0 +1,496 @@
...
+    </div>
+  </div>
+
+  <div class="modal" id="network-ip-settings-dialog"
+       tabindex="-1" role="dialog"
+       data-backdrop="static">
+    <div class="modal-dialog">
+      <div class="modal-content">
+        <div class="modal-header">
+          <h4 class="modal-title">IP Settings</h4>
```
Comment (dperpeet) [inline pkg/networkmanager/interfaces.html]: `translatable`

File: pkg/networkmanager/interfaces.html [hunk 10]
```diff
@@ -0,0 +1,496 @@
...
+    </div>
+  </div>
+
+  <div class="modal" id="network-bond-settings-dialog"
+       tabindex="-1" role="dialog"
+       data-backdrop="static">
+    <div class="modal-dialog">
+      <div class="modal-content">
+        <div class="modal-header">
+          <h4 class="modal-title">Bond Settings</h4>
```
Comment (dperpeet) [inline pkg/networkmanager/interfaces.html]: `translatable`

File: pkg/networkmanager/interfaces.html [hunk 11]
```diff
@@ -0,0 +1,496 @@
...
+    </div>
+  </div>
+
+  <div class="modal" id="network-bridge-settings-dialog"
+       tabindex="-1" role="dialog"
+       data-backdrop="static">
+    <div class="modal-dialog">
+      <div class="modal-content">
+        <div class="modal-header">
+          <h4 class="modal-title">Bridge Settings</h4>
```
Comment (dperpeet) [inline pkg/networkmanager/interfaces.html]: `translatable`

File: pkg/networkmanager/interfaces.html [hunk 12]
```diff
@@ -0,0 +1,496 @@
...
+    </div>
+  </div>
+
+  <div class="modal" id="network-bridgeport-settings-dialog"
+       tabindex="-1" role="dialog"
+       data-backdrop="static">
+    <div class="modal-dialog">
+      <div class="modal-content">
+        <div class="modal-header">
+          <h4 class="modal-title">Bridge Port Settings</h4>
```
Comment (dperpeet) [inline pkg/networkmanager/interfaces.html]: `translatable`

File: pkg/networkmanager/interfaces.html [hunk 13]
```diff
@@ -0,0 +1,496 @@
...
+    </div>
+  </div>
+
+  <div class="modal" id="network-vlan-settings-dialog"
+       tabindex="-1" role="dialog"
+       data-backdrop="static">
+    <div class="modal-dialog">
+      <div class="modal-content">
+        <div class="modal-header">
+          <h4 class="modal-title">VLAN Settings</h4>
```
Comment (dperpeet) [inline pkg/networkmanager/interfaces.html]: `translatable`

File: pkg/networkmanager/interfaces.js [hunk 14]
```diff
@@ -3107,6 +3097,83 @@ function PageNetworkVlanSettings() {
...
+function init() {
+    var model;
+    var overview_page;
+    var interface_page;
+
+    function navigate() {
+        var path = cockpit.location.path;
+
+        if (path.length === 0) {
+            page_hide (interface_page);
```
Comment (dperpeet) [inline pkg/networkmanager/interfaces.js]: see coding guidelines on wiki regarding whitespace in function invocation

**Comment by dperpeet** [comment]
not sure if the `translatable` issues should be addressed separately, but it does seem like an easy change

File: pkg/networkmanager/Makefile.am (hunk 2 above)
Comment (mvollmer) [inline pkg/networkmanager/Makefile.am]: Done.

File: pkg/networkmanager/interfaces.js (hunk 14 above)
Comment (mvollmer) [inline pkg/networkmanager/interfaces.js]: Fixed

**Comment by mvollmer** [comment]
> not sure if the translatable issues should be addressed separately, but it does seem like an easy change

Yep, did that.

**Pull request closed without merging**